pkg/scheduling/requirements.go

```diff
@@ -132,6 +132,19 @@ func (r Requirements) Compatible(requirements Requirements) (errs error) {
 	return multierr.Append(errs, r.Intersects(requirements))
 }
 
+// Compatible ensures the provided requirements can be met.
+func (r Requirements) StrictlyCompatible(requirements Requirements) (errs error) {
+	// All requirements must be defined in the requirements
+	for key := range requirements.Keys() {
+		if operator := requirements.Get(key).Operator(); r.Has(key) || operator == v1.NodeSelectorOpNotIn || operator == v1.NodeSelectorOpDoesNotExist {
```

Comment:
What about keys that are in the Machine requirements, but aren't in the Provisioner requirements? For instance, my Machine Requirements have `Key In Values [a,b,c]` but my Provisioner requirements doesn't have Key. Wouldn't that be incompatible?